.github/workflows/build.yml

```diff
@@ -0,0 +1,76 @@
+name: Build
+
+on:
+  push:
+
+  # Allows manual bumps by API trigger
+  workflow_dispatch:
+    inputs:
+      go_version:
+        description: 'Go version without the "go" prefix'
+        required: false
+
+permissions:
+  contents: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    concurrency:
+      group: gotify-build-lock
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+    
+    - name: Extract input options
+      if: ${{ github.event.inputs.go_version }}
+      run: |
+        export GO_VERSION=${{ github.event.inputs.go_version }}
+
+        echo $GO_VERSION > GO_VERSION
+  
+    - name: Sanity check
+      run: |
+        GO_VERSION=$(cat GO_VERSION)
+        MATCHES=$(curl -fsSL 'https://go.dev/dl/?mode=json&include=all' | jq --arg GO_VERSION "$GO_VERSION" -r ".[] | select (.version == \"go\" + \$GO_VERSION) | .version" | wc -l)
+    
+        if [ "$MATCHES" -ne 1 ]; then
+          echo "Invalid Go version: $GO_VERSION, found $MATCHES matches"
+          exit 1
+        fi
```

Comment:
Do we need the sanity check? I mean the `make build` will also fail, if the version is invalid.

Reply:
If you put 1.23 it will build 1.23.2 too when you want 1.23.2

Reply:
Good point :+1: